packages/design-system/src/components/web-components/ds-modal-dialog/ds-modal-dialog.stories.tsx

```diff
@@ -124,27 +122,23 @@ const Template = (args) => {
   useEffect(() => {
     const modal = document.querySelector('ds-modal-dialog');
     const toggleButton = document.getElementById('modal-toggle');
-    const closeButton = document.querySelector('button[aria-label="Close modal dialog"]');
-    const form = document.querySelector('form[method=dialog]');
 
-    const toggleModal = () => {
-      if (modal.getAttribute('is-open') === 'true') {
-        modal?.setAttribute('is-open', 'false');
-      } else {
-        modal?.setAttribute('is-open', 'true');
-      }
+    const openModal = () => {
+      modal?.setAttribute('is-open', 'true');
     };
 
-    modal?.addEventListener('ds-exit', (event) => action('ds-exit')(event));
-    toggleButton?.addEventListener('click', toggleModal);
-    form?.addEventListener('submit', toggleModal);
-    closeButton?.addEventListener('click', toggleModal);
+    modal?.addEventListener('ds-exit', (event) => {
+      action('ds-exit')(event);
+      modal?.setAttribute('is-open', 'false');
+    });
+    toggleButton?.addEventListener('click', openModal);
 
     return () => {
-      modal?.removeEventListener('ds-exit', (event) => action('ds-exit')(event));
-      toggleButton?.removeEventListener('click', toggleModal);
-      form?.removeEventListener('submit', toggleModal);
-      closeButton?.removeEventListener('click', toggleModal);
+      modal?.removeEventListener('ds-exit', (event) => {
+        action('ds-exit')(event);
+        modal?.setAttribute('is-open', 'false');
+      });
```

Comment:
I did this in my troubleshooting branch, but I want to flag it here because these anonymous functions are two different objects, so it won't actually remove the original event listener added on line 130.